starport/services/network/join.go

```diff
@@ -170,26 +181,42 @@ func (n Network) sendValidatorRequest(
 }
 
 // hasValidator verify if the validator already exist into the SPN store
-func (n Network) hasValidator(ctx context.Context, launchID uint64, address string) bool {
+func (n Network) hasValidator(ctx context.Context, launchID uint64, address string) (bool, error) {
 	_, err := launchtypes.NewQueryClient(n.cosmos.Context).GenesisValidator(ctx, &launchtypes.QueryGetGenesisValidatorRequest{
 		LaunchID: launchID,
 		Address:  address,
 	})
-	return err == nil
+	err = sdkerror.Unwrap(err)
+	if err == sdkerror.ErrInvalidRequest {
+		return false, nil
+	} else if err != nil {
+		return false, err
+	}
+	return true, nil
 }
 
 // hasAccount verify if the account already exist into the SPN store
-func (n Network) hasAccount(ctx context.Context, launchID uint64, address string) bool {
+func (n Network) hasAccount(ctx context.Context, launchID uint64, address string) (bool, error) {
 	_, err := launchtypes.NewQueryClient(n.cosmos.Context).VestingAccount(ctx, &launchtypes.QueryGetVestingAccountRequest{
 		LaunchID: launchID,
 		Address:  address,
 	})
-	if err == nil {
-		return true
+	err = sdkerror.Unwrap(err)
+	if err != nil && err != sdkerror.ErrInvalidRequest {
+		return false, err
+	} else if err == nil {
+		return true, nil
 	}
+
 	_, err = launchtypes.NewQueryClient(n.cosmos.Context).GenesisAccount(ctx, &launchtypes.QueryGetGenesisAccountRequest{
 		LaunchID: launchID,
 		Address:  address,
 	})
-	return err == nil
+	err = sdkerror.Unwrap(err)
+	if err != nil && err != sdkerror.ErrInvalidRequest {
+		return false, err
+	} else if err == nil {
```

Comment:
`err == nil` is redundant check. Err checked previously against nil. If `err != nil` fails then err is nil.

Reply:
the first condition checks `if err != nil && err != sdkerror.ErrInvalidRequest` not only the `err != nil`. It's not redundancy, but maybe we can make more clear in the code

Reply:
Oh yeah I missed `&&` sorry. But if `err != nil && err != sdkerror.ErrInvalidRequest` and `err == nil` fail it means err is not handled?

Reply:
Why not having
```
err = sdkerror.Unwrap(err)
	if err == sdkerror.ErrInvalidRequest {
		return false, nil
	} else if err != nil {
		return false, err
	}
	return true, nil
```
As above (that I find clearer), is there a difference?

Reply:
Yep that one is the best. The main issue of current statement I find it hard to read and understand. It captures more attention than it should.